helper/versions/version.go

```diff
@@ -52,3 +53,19 @@ func GetBuiltinVersion(pluginType consts.PluginType, pluginName string) string {
 
 	return DefaultBuiltinVersion
 }
+
+func IsBuiltinVersion(v string) bool {
+	semanticVersion, err := semver.NewSemver(v)
+	if err != nil {
+		return false
+	}
+
+	metadataIdentifiers := strings.Split(semanticVersion.Metadata(), ".")
+	for _, identifier := range metadataIdentifiers {
+		if identifier == "builtin" {
+			return true
```

Comment:
nit: I have seen this string literal duplicated in a few places, wonder if it would be good to factor it out to an exported constant.